python/hsfs/storage_connector.py

```diff
@@ -336,300 +294,220 @@ def iam_role(self):
-            )
+        return self._url.replace("https://", "").replace(".snowflakecomputing.com", "")
 
     @property
-    def sf_options(self):
+    def options(self):
         """Additional options for the Snowflake storage connector"""
-        if self._storage_connector_type.upper() == self.SNOWFLAKE:
-            return self._sf_options
-        else:
-            raise Exception(
-                "sf_options are not supported for connector "
-                + self._storage_connector_type
-            )
-
-    def spark_options(self):
-        """Return prepared options to be passed to Spark, based on the additional
-        arguments.
-        """
-        if self._storage_connector_type.upper() == self.JDBC:
-            args = [arg.split("=") for arg in self._arguments.split(",")]
-
-            options = {a[0]: a[1] for a in args}
-            options["url"] = self._connection_string
-
-            return options
-        elif self._storage_connector_type.upper() == self.REDSHIFT:
-            connstr = (
-                "jdbc:redshift://"
-                + self._cluster_identifier
-                + "."
-                + self._database_endpoint
-                + ":"
-                + str(self._database_port)
-                + "/"
-                + self._database_name
-            )
-            if self._arguments is not None:
-                connstr = connstr + "?" + self._arguments
-            props = {
-                "url": connstr,
-                "driver": self._database_driver,
-                "user": self._database_user_name,
-                "password": self._database_password,
-            }
-            if self._table_name is not None:
-                props["dbtable"] = self._table_name
-            return props
-        elif self._storage_connector_type.upper() == self.ADLS:
-            return self._spark_options
-        elif self._storage_connector_type.upper() == self.SNOWFLAKE:
-            props = self._sf_options
-            props["sfURL"] = self._url
-            props["sfSchema"] = self._schema
-            props["sfDatabase"] = self._database
-            props["sfUser"] = self._user
-            if self._password is not None:
-                props["sfPassword"] = self._password
-            else:
-                props["sfAuthenticator"] = "oauth"
-                props["sfToken"] = self._token
-            if self._warehouse is not None:
-                props["sfWarehouse"] = self._warehouse
-            if self._role is not None:
-                props["sfRole"] = self._role
-            if self._table is not None:
-                props["dbtable"] = self._table
-
-            return props
-        else:
-            raise Exception(
-                "Spark options are not supported for connector "
-                + self._storage_connector_type
-            )
+        return self._options
 
     def snowflake_connector_options(self):
```

Comment:
Is this method being used anywhere?

Reply:
Okay, I know now what it's for, it prepares the options to connect with the `snowflake.connector` python library. the arguments are slightly different than the spark options. so it makes sense to have it. Added documentation with an exmple for it in the docstring